prime-router/src/main/kotlin/transport/SftpTransport.kt

```diff
@@ -1,58 +1,90 @@
+        header: DatabaseAccess.Header,
+        contents: ByteArray
+    ): Boolean {
 
-        val fileDir = service.transport.filePath.removeSuffix("/")
+        val sftpTransportType = transportType as SFTPTransportType
 
-        // TODO - determine what the filename should be
-        val path = "$fileDir/${service.fullName.replace('.', '-')}-${header.task.reportId}.csv"
-        val host: String = service.transport.host
-        val port: String = service.transport.port
+        val (user, pass) = lookupCredentials(orgName)
 
-        val jsch = JSch()
-        val jschSession = jsch.getSession(user, host, port.toInt())
-        val config = Properties()
-        config.put("StrictHostKeyChecking", "no")
-        config.put("PreferredAuthentications", "password")
-        jschSession.setConfig(config)
-        jschSession.setPassword(pass)
+        val fileDir = sftpTransportType.filePath.removeSuffix("/")
 
-        jschSession.connect()
-        val channelSftp = jschSession.openChannel("sftp") as ChannelSftp
+        val path = "$fileDir/${orgName.replace('.', '-')}-${header.task.reportId}.csv"
+        val host: String = sftpTransportType.host
+        val port: String = sftpTransportType.port
 
-        var success = false
+        var success: Boolean
 
-        try {
-            channelSftp.connect()
-            channelSftp.put(contents.inputStream(), path, ChannelSftp.OVERWRITE)
+            uploadFile(host, port, user, pass, path, contents )
             success = true
-        } finally {
-            channelSftp.disconnect()
-        }
 
         return success
     }
 
-    private fun lookupCredentials(service: OrganizationService): Pair<String, String> {
+    private fun lookupCredentials(orgName: String): Pair<String, String> {
 
-        val envVarLabel = service.fullName.replace(".", "__").replace('-', '_').toUpperCase()
+        val envVarLabel = orgName.replace(".", "__").replace('-', '_').toUpperCase()
 
         val user = System.getenv("${envVarLabel}__USER") ?: ""
         val pass = System.getenv("${envVarLabel}__PASS") ?: ""
 
         if (user.isNullOrBlank() || pass.isNullOrBlank())
-            error("Unable to find SFTP credentials for ${service.fullName}")
+            error("Unable to find SFTP credentials for $orgName")
 
         return Pair(user, pass)
     }
+
+    private fun setupSshj(
+        host: String,
+        port: String,
+        user: String,
+        pass: String
+    ): SSHClient {
+
+        val client:SSHClient = SSHClient();
+        client.addHostKeyVerifier(PromiscuousVerifier());
+        client.connect(host,port.toInt());
+        client.authPassword(user, pass);
+        return client;
+    }
+
+    private fun uploadFile(
+        host: String,
+        port: String,
+        user: String,
+        pass: String,
+        path: String,
+        contents: ByteArray
+    ) {
+        val sshClient: SSHClient = setupSshj(host, port, user, pass);
```

Comment:
Suggest that this block should protected against exceptions being thrown. Either a `sshClient.use { }` block since sshClient implements closable or a try-catch block if disconnect is really needed.

Reply:
Changed to use blocks for sshClient and sftpClient